crates/core/src/test_utils/bin/dump-katana.rs

```diff
@@ -24,43 +24,40 @@ async fn main() {
         })
         .await;
 
-    tokio::task::spawn_blocking(move || {
-        // Get a serializable state for the sequencer
-        let sequencer = test_context.sequencer();
-        let dump_state = sequencer
-            .sequencer
-            .backend
-            .state
-            .blocking_write()
-            .dump_state()
-            .expect("Failed to call dump_state on Katana state");
+    // Get a serializable state for the sequencer
+    let sequencer = test_context.sequencer();
+    let dump_state = sequencer
+        .sequencer
```

Comment:
so this is `test_context.sequencer().sequencer.sequencer` ? looks like something went wrong

Reply:
Will create an issue for this in code smell 👍